assessments/CLI_Circle/earthmover.yml

```diff
@@ -0,0 +1,589 @@
+      'name': 'Motivation to Read - En',
+      'objective_assessments': []
+    },
+    'cpm_admin_mode_sp': {
+      'name': 'Administration Mode',
+      'objective_assessments': []
+    }
+  }
+%}
+
+# Stack the dictionaries together into one large dictionary that we can iterate through bellow.
+{% set subjects = subjects_en.copy() %}
+{% set _ = subjects.update(subjects_spa) %}
+
+
+# PREP-STEP 3: CREATE LIST OF SUBJECTS WITHOUT CUT POINT VALUES (PERFORMANCE INDICATORS)
+# Some subjects don't carry CP values, and we need to know which ones don't so that we can create empty performance columns for those.
+# If we don't create empty columns, the "keep columns" operation below won't finish successfully.
+{% set non_cp_columns = [      
+    'approaches_to_learning_exp_v2_creativity_dramatic_play_v1',
+    'approaches_to_learning_exp_v2_flexibility_v1',
+    'approaches_to_learning_exp_v2_initiative_curiosity_v1',
+    'cpm_admin_mode',
+    'math_opt_v1_patterns_v1',
+    'math_opt_v1_real_world_v1',
+    'physical_dev_health_v1_fine_visual_motor_v1',
+    'physical_dev_health_v1_gross_motor_v1',
+    'physical_dev_health_v1_health_v1',
+    'social_emotional_v1_approaches_to_learning_v1',
+    'social_emotional_v1_self_care_v1',
+    'social_emotional_v1_emotion_behavior_v1',
+    'social_emotional_v1_classroom_community_safety_v1',
+    'social_emotional_v1_positive_social_v1',
+    'approaches_to_learning_extended_spa_v1_initiative_curiosity_spa_v1',
+    'approaches_to_learning_extended_spa_v1_flexibility_spa_v1',
+    'approaches_to_learning_extended_spa_v1_creativity_dramatic_play_spa_v1',
+    'cpm_admin_mode_sp',
+    'math_opt_spa_v1_patterns_spa_v1',
+    'math_opt_spa_v1_real_world_spa_v1',
+    'physical_dev_health_spa_v1_fine_visual_motor_spa_v1',
+    'physical_dev_health_spa_v1_gross_motor_spa_v1',
+    'physical_dev_health_spa_v1_health_spa_v1',
+    'social_emotional_spa_v2_positive_social_spa_v1',
+    'social_emotional_spa_v2_classroom_community_safety_spa_v1',
+    'social_emotional_spa_v2_emotion_behavior_spa_v1',
+    'social_emotional_spa_v2_self_care_spa_v1',
+    'social_emotional_spa_v2_approaches_to_learning_spa_v1'
+] 
+%}
+
+
+
+# TRANSFORMATION-STEP 1: CONVERT COLUMNS TO SNAKE CASE
+# Some columns in the data come through with spaces or using "-" symbols instead of "_" as separators.
+# We need to make sure all columns are snake cased before we begin working on the data. 
+# We can also build our student_wave_id here, which will help us identify a student and wave row.
+transformations:
+  cli_circle_student_assessment:
+    source: $sources.cli_circle_input
+    operations:
+      - operation: snake_case_columns 
+      - operation: add_columns
+        columns:
+          # do we even need this? the wave column comes with values that are not snake cased (e.g. "wave 1") so we might need to get rid of this?
+          student_wave_id: '{%raw%}{{wave}}_{{school_year}}_{{local_student_id}}{%endraw%}'
```

Comment:
how can we confirm that this will be unique? can we look at the files we have to start?

I wonder if it's safer to use student_state_id or maybe student_engage_id instead of local?

Reply:
Let's use student engage ID since it is always filled in -- check if a student has taken all assessments twice (ask CLI)

Reply:
have you done any checks on student duplicates, and/or forwarded this Q to Christian?

Reply:
see [this](https://github.com/edanalytics/earthmover_edfi_bundles/pull/32#discussion_r1447939195) comment